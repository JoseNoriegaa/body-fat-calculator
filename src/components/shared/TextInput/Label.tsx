/* eslint-disable jsx-a11y/label-has-associated-control */
// External dependencies
import React from 'react';

// Internal dependencies
import useStyles from './Label.styles';

// Types & Interfaces
interface ILabelProps {
  htmlFor?: string;
}

function Label({ children, htmlFor }: React.PropsWithChildren<ILabelProps>) {
  const classes = useStyles();

  return (
    <label htmlFor={htmlFor} className={classes.label}>
      {children}
    </label>
  );
}

export default Label;
