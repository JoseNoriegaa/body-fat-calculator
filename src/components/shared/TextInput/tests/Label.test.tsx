// External dependencies
import { render, screen } from '@testing-library/react';

// Internal dependencies
import Label from '../Label';

describe('<Label />', () => {
  it('should match snapshot', () => {
    const { asFragment } = render(<Label htmlFor="some_input">form field</Label>);

    const element = screen.getByText('form field');
    expect(element).toBeInTheDocument();
    expect(element).toHaveAttribute('for', 'some_input');
    expect(asFragment()).toMatchSnapshot();
  });
});
