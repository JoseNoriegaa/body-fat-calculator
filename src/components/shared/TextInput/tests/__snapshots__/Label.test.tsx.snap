// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Label /> should match snapshot 1`] = `
<DocumentFragment>
  <label
    class="label-0-2-1"
    for="some_input"
  >
    form field
  </label>
</DocumentFragment>
`;
